github-release

```diff
@@ -164,12 +164,36 @@ if [ "$PREVIEW" = true ]
   exit
 fi
 
-# Find remote repository
-REPOSITORY=$(git config --get remote.origin.url | sed -E 's/https?:\/\/(www.)?github.com\/|git@github.com:([A-Za-z]+\/[A-Za-z]+)/\2/')
+
+# obtain remote repository details
+function get_repository_details
+{
+  read -r -p 'Repository URL > ' REPO_URL
+  read -r -p 'Default Branch > ' REPO_BRANCH
+}
 
 GREY_COLOR="\033[0;37m"
 NO_COLOR="\033[0m"
 
+# if origin isn't set
+if ! git config --get remote.origin.url
+  then
+    echo -e "Remote ${GREY_COLOR}origin${NO_COLOR} missing"
+    echo -e "You need to specify the remote repository manually"
+    get_repository_details
+    # while remote isn't valid, ask for details again
+    while [[ ! $(timeout 30 git ls-remote -q "$REPO_URL") ]]
```

Comment:
* I've set the timeout to be 30s, opinions ?
  - I've done this since `ls-remote` may take too much time when the Repo URL isn't valid

* I've chosen the 3rd Option :laughing:

Reply:
I like the third option 😉 . A couple last things before I merge this:

* `timeout` is not universal, does not exist on MacOS per default for example. I would suggest we leave it up to the user and just read `REPO_URL` and `REPO_BRANCH` once. I want to improve error messages on failed API requests anyway ( #24  ), so this case will be covered there. Basically remove the while loop entirely.

* Currently `if ! git config --get remote.origin.url` prints the result to the console, which is not what we want. Using `[ -z "$(git config --get remote.origin.url)" ]` fixes this for me. I'm not 100% sure about the exact difference between the two, but on MacOS there is no output on the second one, which is more appropriate here.

Reply:
Ok, so the `ls-remote` part would be removed.
Fine with me, since you said it'd taken care on sending an API request

I'd come up with the **First Option** essentially soon